src/scheduler.cc

```diff
@@ -456,7 +460,28 @@ std::pair<WorkerId, ObjStoreId> SchedulerService::register_worker(const std::str
     } else {
       (*workers)[workerid].current_task = NO_OPERATION;
     }
+
+    if (!is_driver) {
+      // Import remote functions on the worker.
+      for (int i = 0; i < exported_functions.size(); ++i) {
+        ClientContext import_context;
+        ImportFunctionRequest import_request;
+        import_request.mutable_function()->CopyFrom(*exported_functions[i].get());
+        ImportFunctionReply import_reply;
+        (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
+      }
+
+      // Import reusable variables on the worker.
+      for (int i = 0; i < exported_reusable_variables.size(); ++i) {
+        ClientContext import_context;
+        ImportReusableVariableRequest import_request;
+        import_request.mutable_reusable_variable()->CopyFrom(*exported_functions[i].get());
+        AckReply import_reply;
+        (*workers)[i].worker_stub->ImportReusableVariable(&import_context, import_request, &import_reply);
+      }
+    }
```

Comment:
Perhaps we should have separate methods
```
import_function_on_worker(workerid, functionid)
import_reusable_variable_on_worker(workerid, variableid)
```
which would be used here in `register_worker` as well as in `ExportFunction` and `ExportReusableVariable`.

This would require a little bit of reorganization, e.g., should think about the relation between `fntable_` and `exported_functions`